widget/entry.go

```diff
@@ -378,124 +395,131 @@ func (e *Entry) TypedKey(key *fyne.KeyEvent) {
 
 	switch key.Name {
 	case fyne.KeyBackspace:
-		e.RLock()
-		isEmpty := provider.len() == 0 || (e.CursorColumn == 0 && e.CursorRow == 0)
-		e.RUnlock()
+		isEmpty := e.GetField(func() interface{} {
+			return provider.len() == 0 || (e.CursorColumn == 0 && e.CursorRow == 0)
+		}).(bool)
 		if isEmpty {
 			return
 		}
-		pos := e.cursorTextPos()
-		e.Lock()
-		provider.deleteFromTo(pos-1, pos)
-		e.CursorRow, e.CursorColumn = e.rowColFromTextPos(pos - 1)
-		e.Unlock()
+		e.SetFields(func() {
+			pos := e.cursorTextPos()
+			provider.deleteFromTo(pos-1, pos)
+			e.CursorRow, e.CursorColumn = e.rowColFromTextPos(pos - 1)
+		})
 	case fyne.KeyDelete:
 		pos := e.cursorTextPos()
 		if provider.len() == 0 || pos == provider.len() {
 			return
 		}
-		provider.deleteFromTo(pos, pos+1)
+		e.SetFields(func() {
+			provider.deleteFromTo(pos, pos+1)
+		})
 	case fyne.KeyReturn, fyne.KeyEnter:
-		if !e.MultiLine {
+		if e.GetField(func() interface{} {
+			return !e.MultiLine
+		}).(bool) {
 			return
 		}
-		provider.insertAt(e.cursorTextPos(), []rune("\n"))
-		e.Lock()
-		e.CursorColumn = 0
-		e.CursorRow++
-		e.Unlock()
+		e.SetFields(func() {
+			provider.insertAt(e.cursorTextPos(), []rune("\n"))
+			e.CursorColumn = 0
+			e.CursorRow++
+		})
 	case fyne.KeyUp:
-		if !e.MultiLine {
+		if e.GetField(func() interface{} {
+			return !e.MultiLine
+		}).(bool) {
```

Comment:
This is repeated a few times, perhaps add "func (e *Entry) IsMultiline() bool" ?

Reply:
I'll just grab it at the top, seems seasier